Foundation/NSCompoundPredicate.swift

```diff
@@ -44,11 +44,17 @@ public class NSCompoundPredicate : NSPredicate {
     }
 
     override public func evaluateWithObject(object: AnyObject?, substitutionVariables bindings: [String : AnyObject]?) -> Bool {
+        let subpredicates = self.subpredicates as! [NSPredicate]
+        
         switch compoundPredicateType {
         case .AndPredicateType:
-            return subpredicates.reduce(true, combine: { $0 && $1.evaluateWithObject(object, substitutionVariables: bindings) })
+            return subpredicates.reduce(true, combine: {
+                $0 && $1.evaluateWithObject(object, substitutionVariables: bindings)
+            })
         case .OrPredicateType:
-            return subpredicates.reduce(false, combine: { $0 || $1.evaluateWithObject(object, substitutionVariables: bindings) })
+            return subpredicates.reduce(false, combine: {
+                $0 || $1.evaluateWithObject(object, substitutionVariables: bindings)
+            })
```

Comment:
Using `reduce` doesn't maintain the short-circuiting behavior that the ObjC Foundation version exhibits. As a test, the following will not crash, indicating that `NSCompoundPredicate` is short-circuiting predicate evaluation once it knows that it will return `false`:

    let bOK = NSPredicate(value: false)
    let bCrash = NSPredicate(block: { _ in
        fatalError("whoops")
        return false
    })

    let both = NSCompoundPredicate(andPredicateWithSubpredicates: [bOK, bCrash])
    let eval = both.evaluateWithObject("hi") // evaluates to false